### DISCUSSION TITLE
How to proceed to validate the Pains an Gains?

### DISCUSSION THREAD

**Discussion by manuGil** [ITC-CRIB/jupyter-fairly#8] [Technology options and analysis]
#  Which approach to follow to validate the pains and gains by end-users?

### Some initial ideas:
- A focus group 
- A survey

**Comment by jurra** [comment]
**@manuGil and @girgink I see two steps as we have discussed in the meeting, with regard to the survey**

Step 1: Formulate the tasks, pains and gains into user stories that we can present in a survey. I found [this resource](https://www.aha.io/roadmapping/guide/requirements-management/what-is-a-good-feature-or-user-story-template#:~:text=What%20is%20a%20user%20story,want%20or%20need%20to%20do.) on how to formulate user stories. They provide templates from simple ot more complex. Now that you have distilled the tasks, pains and gains into a single list, we can formulate them in such a format. We could use this list as statements to be presented by users. 

Step 2: Build the survey by adding all these stories where people can rate them from: not relevant, somewhat relevant, very relevant. 

**Steps that can follow up towards implementing stories/features**

Step 3: Prioritize based on desirability and feasibility. I am not sure we are going to get statistically relevant results very soon, but the list can give us a nice indication of things to focus on.

Step 4: Update our backlog of stories and development plan based on this results. Once we have a list of well formulated stories and technical hypothesis to satisfy them, it is easy to manage the list of things we want to try. For instance we can decide on a first sprint what can we focus on ;)

**Reply by jurra** [reply]
About the focus group, I am not so sure how to proceed, or how to facilitate it and why we want to do it at this point.

**Reply by connieclare2802** [reply]
Hi @jurra, @manuGil, @girgink and @roelj 

I have been trying to catch up with the discussions and have a suggestion for you to think about (I appreciate I wasn't present in the last meeting so I may be missing important aspects of your discussion here...) 

Personally, I would be inclined to start involving the community by organising a small focus group of maybe 6 people or so before circulating a survey. Here are my reasons for this: 
- Inviting personnel (data stewards, data managers, RSEs and researchers) who have some understanding of JupyterLab, data repositories (4TU) and the research workflow means that we can quickly collect feedback on our initial ideas and refine them for the survey. I see this as a two-step process as the focus group can help you construct a survey. A focus group can allow us to uncover hidden information that can be used to inform our survey design.
- The focus group becomes an initial 'advisory/sounding board' that you can go back to for feedback at the later stages of the project; they may become core community members who can help you recruit other community members to test and deploy the product in the future.
- A survey is a very useful way to collect quantifiable and easy to analyse data but they are less personal and interactive. In my experience, you have to have clear goals and objectives when sending a survey; you have to know what to ask and you have to identify your target audience. You also need large numbers of respondents in order to draw significant conclusions. 
- A focus group provides more in-depth discussion and qualitative information which would be useful at this early stage off the project. At this stage, I think we are looking for 'quality over quantity' to direct the technical development. 

These are just my thoughts - Happy to discuss if you like of course :)

**Reply by jurra** [reply]
Connie I understand where you are coming from. But we are already kind of a focus group. Specially considering that we all have some kind of research or academic related background. I am more inclined towards a survey I  will explain more on this later...

**Reply by connieclare2802** [reply]
Thanks Jose. I appreciate your comments and I think a survey will certainly be required at some stage. However, I do not believe we are a focus group; we are the development team. We require feedback, advice and verification of our ideas by others who can help us. 

The decision about the next steps taken to involve the community should be made as a team so perhaps you can cast a vote in the next meeting about how to proceed so that everyone in the team is happy with the decision?

**Reply by jurra** [reply]
Hello @connieclare2802 I explain a bit more what I mean.
I agree we need feedback and verification, but of what at this point? And from who? For example having a clear list of formulated stories, where we clearly specify a need from the user, and what he/she wants to do, is a concrete thing to get feedback on. 

Community engagement takes place differently depending on the maturity of the project, the point is that it is important to have feedback on something concrete. I am in favor of a focus group given a certain format, and concrete things to get feedback upon.

The focus group of certain type of users/roles/stakeholders can come after as well for instance. The focus group can also be created out of the results of a survey for instance.  

So a general question about your proposal for the focus group. **What feedback do you think we need at this point? Then, could you explain how you think we can get it?** That would lead to the concrete course of action which can be a focus group, or something else.

We can also reflect on how much time we need for this, in contrast to other things we can do in parallel.

**Reply by jurra** [reply]
With regard to your comment on whether we are users or developers, we are both. It is quite common that some users progress towards developing tools, this is how the proposal has come about. 

This is what I meant when I mentioned that we are somewhat a little focus group since all of us have had some usage experience with archiving.

**Reply by connieclare2802** [reply]
@jurra - The same questions apply whether you run a focus group or a survey. Since we are in the planning phase, I agree that the 'user stories' are a good place to start. I would have held an initial focus group with 4TU data stewards to:
1) present the project aims and objectives 
2) pitch the user stories 
3) gather feedback on our project plan, and confirm whether our understanding of the pains/gains/tasks are accurate. 

This way, the 4TU data stewards get to learn about the project from the beginning and may have better understanding when they are invited to participate at later stages for testing and deployment. 

As I mentioned, I am not against sending a survey, so if @heijer, @manuGil and @girgink agree this is the best way to proceed then I'm happy with that. The same questions you asked me about the focus group also apply to sending a survey: 

- What feedback do you think we need at this point? Are the user stories accurate and concrete enough to get feedback on?
- Who is the target audience for the survey? 
- How many respondents do you need/want to complete the survey to gather useful feedback?
- How will you ensure respondents have sufficient background knowledge of the project to complete the survey?
- What questions will you ask?

I imagine you already have good ideas for how to engage the community though a survey, so we can discuss this next week as a group :)

**Reply by jurra** [reply]
Hi @connieclare2802 thanks again for the remarks
> What feedback do you think we need at this point? 

Honestly, the most important feedback at this point is alignment on what problems we want to focus on, and some very basic consent on what is interesting and valuable to work on at the moment. I am not expecting more feedback than this in such an early stage of a project.

> Are the user stories accurate and concrete enough to get feedback on?

They are concrete enough for us to be able to talk about the project and understand each others point of view. They are also formulated in a way that is understandable for users and developers, and they help to scope and focus. That is their main goal.

> How many respondents do you need/want to complete the survey to gather useful feedback?

I am not thinking at this point about a big survey, instead a little one that we can try with few people and learn from that. Then evolve and invite more to get feedback.

> How will you ensure respondents have sufficient background knowledge of the project to complete the survey?

By having a clear persona definition (A researcher that often needs to archive and publish their data). A researcher that also do this so often and wants to teach other colleagues to do so, therefore wants t make it easy and simple (That is Serkan I think ;), among others)

> What questions will you ask?

How relevant is this goal for you when you are in the process of publishing data (From 1 to 5)(not relevant, very relevant). An then put the users story. This is my first simple idea about it. Perhaps we have to simplify them a bit to make them survey friendly.

Besides this discussion. There is another fast track way of doing things which is also valid for this kind of maturity in such a project.
You simply tackle some of your hypothesis that are already sensible, logical, and prototype a solution that addresses some of these. Then comeback to the community and get some feedback and keep iterating. This is my preferred way in such an early stage project, with so few resources, simply due to time limitations.

**Comment by girgink** [comment]
Initial user stories are now available in the [user stories document]( https://github.com/ITC-CRIB/JupyterFAIR/blob/main/docs/user_stories.md). Can you please have a look at it?

Regarding how to get feedback, I think we can combine both approaches. We can organize a focus group to get a targeted and quick feedback, but at the same time ask the community for their feedback through social media and our communication channels. This may allow us to start initial development with the focus group feedback and fine tune it in time based on the community feedback.

**Reply by jurra** [reply]
Thanks for sharing Serkan will look into it tomorrow in the afternoon.
